src/internal_events/tcp.rs

```diff
@@ -0,0 +1,135 @@
+use super::InternalEvent;
+use metrics::counter;
+
+#[derive(Debug)]
+pub struct TcpConnectionEstablished {
+    pub peer_addr: Option<std::net::SocketAddr>,
+}
+
+impl InternalEvent for TcpConnectionEstablished {
+    fn emit_logs(&self) {
+        if let Some(peer_addr) = self.peer_addr {
+            debug!(message = "connected", %peer_addr);
+        } else {
+            debug!(message = "connected", peer_addr = "unknown");
+        }
+    }
+
+    fn emit_metrics(&self) {
+        counter!("tcp_connections_established", 1,
+            "component_kind" => "sink",
```

Comment:
I'm assuming these don't have a type because they can be crated from many different types of sinks and we don't have that metadata here?

Reply:
Correct. This is where we'll need to lean on `tracing` metadata.